ecommerce_integrations/public/js/unicommerce/pick_list.js

```diff
@@ -0,0 +1,53 @@
+frappe.ui.form.on('Pick List', {
+    refresh(frm){
+        if (frm.doc.order_details){
+            cur_frm.add_custom_button(__('Generate Invoice'), () => frm.trigger('generate_invoice'))
+        }
+    },
+    generate_invoice(frm){
+        let selected_so = []
+        var tbl = cur_frm.doc.order_details || [];
+            for(var i = 0; i < tbl.length; i++) {
+                selected_so.push(tbl[i].sales_order)
+            }
+            let sales_orders = [];
+            let so_item_list = [];
+        const warehouse_allocation = {};
+        selected_so.forEach(function(so) {
+                        const item_details = frm.doc.locations.map((item) => {
+                        if (item.sales_order == so && item.picked_qty > 0){
+                                so_item_list.push({so_item:item.sales_order_item,
+                                                qty:item.qty
+                                          });
+                                        return {
+                                                sales_order_row: item.sales_order_item,
+                                                item_code: item.item_code,
+                                                warehouse: item.warehouse,
+                                                shelf:item.shelf
+                                        }
+                                        }
+                                      else{
+                                          return {}
+                                      }
+                        });
+                        sales_orders.push(so);
+                            warehouse_allocation[so] = item_details.filter(value => Object.keys(value).length !== 0);
+                    });
+                    frappe.call({
+			method: 'ecommerce_integrations.unicommerce.pick_list.validate_partial_picking',
```

Comment:
These validations should ideally be done on server side only?

Reply:
Done